src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -728,6 +728,45 @@ public function readFile(string $filename): string
         return $content;
     }
 
+    /**
+     * Reads a file in fixed-size chunks and yields them one at a time.
+     *
+     * @param string $filename The full path to the file
+     *
+     * @return \Generator<string> Yields file content as strings in chunks
+     *
+     * @throws IOException If the file cannot be opened or read, or if it's a directory
```

Comment:
this is wrong in the current implementation. This exception is not thrown when calling this method, but when starting the iteration (your test is hiding that by using `iterator_to_array`).
To properly perform the validation synchronously (which is easier for error handling and documentation), you would need to move the iteration to a private method (defining the generator) while the validation runs in the public method before calling that private method. We use that approach in `symfony/cache` for instance (where we are required to validate keys synchronously to respect PSR-6)